src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/4.diff.sql

```diff
@@ -0,0 +1,410 @@
+
+/*************************************************************
+    Resource table
+**************************************************************/
+
+ALTER TABLE dbo.Resource
+    ADD IsRawResourceMetaSet bit NOT NULL DEFAULT 0
+GO
+
+--
+-- STORED PROCEDURE
+--     ReadResource
+--
+-- DESCRIPTION
+--     Reads a single resource, optionally a specific version of the resource.
+--
+-- PARAMETERS
+--     @resourceTypeId
+--         * The ID of the resource type (See ResourceType table)
+--     @resourceId
+--         * The resource ID
+--     @version
+--         * A specific version of the resource. If null, returns the latest version.
+-- RETURN VALUE
+--         A result set with 0 or 1 rows.
+--
+ALTER PROCEDURE dbo.ReadResource
+    @resourceTypeId smallint,
+    @resourceId varchar(64),
+    @version int = NULL
+AS
+    SET NOCOUNT ON
+
+    IF (@version IS NULL) BEGIN
+        SELECT ResourceSurrogateId, Version, IsDeleted, IsHistory, IsRawResourceMetaSet, RawResource
+        FROM dbo.Resource
+        WHERE ResourceTypeId = @resourceTypeId AND ResourceId = @resourceId AND IsHistory = 0
+    END
+    ELSE BEGIN
+        SELECT ResourceSurrogateId, Version, IsDeleted, IsHistory, IsRawResourceMetaSet, RawResource
```

Comment:
I think we have a problem here. By "altering" this stored procedure in this way, where the output schema changes, we would break existing clients that are using the "V3" schema. 
(Apologies, we have not gone through all the schema upgrade scenarios yet).
I can think of two ways we could deal with this.

1.  We create a new stored procedure called something like `dbo.ReadResource_4`. Ideally, the codegen would strip off the suffix. I think the integration test that verifies upgrade schema compatibility would fail because of the extra stored procedure left over from the previous version.
2. We keep this an `ALTER PROCEDURE` but move `IsRawResourceMetaSet` to the last column in the select statement. Then the code that reads it would need to read that column after reading the RawResource stream (if of course the schema version > 3). I think we want to do that outside of the ReadRow call because we do not want the the SqlDataReader to buffer the entire RawResource.

It seems like in the general case we will need to support (1), but we can probably get away with (2) here.